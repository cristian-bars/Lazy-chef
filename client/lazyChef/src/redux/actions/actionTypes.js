export default {
  LOAD_RECIPES: 'LOAD_RECIPES',
  DELETE_RECIPE: 'DELETE_RECIPE',
  ADD_RECIPE: 'ADD_RECIPE',
  UPDATE_RECIPE: 'UPDATE_RECIPE',
  LOAD_RECIPE: 'LOAD_RECIPE',

  LOAD_USERS: 'LOAD_USERS',
  DELETE_USER: 'DELETE_USER',
  ADD_USER: 'ADD_USER',
  UPDATE_USER: 'UPDATE_USER',
  LOAD_USER: 'LOAD_USER',
};
