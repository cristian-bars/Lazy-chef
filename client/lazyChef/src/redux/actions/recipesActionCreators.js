import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/recipes';
const url = 'http://192.168.0.29:2022/recipes';
//const url = 'http://192.168.1.130:2022/recipes';

export function loadRecipes() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_RECIPES,
        recipes: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPES_ERROR',
      });
    }
  };
}

export function addRecipe(recipe) {
  return {
    type: actionTypes.ADD_RECIPE,
    recipe,
  };
}

export function deleteRecipe(recipeId) {
  return async dispatch => {
    await axios.delete(`${url}/${recipeId.id}`);
    dispatch({
      type: actionTypes.DELETE_RECIPE,
      recipeId,
    });
  };
}

export function updateRecipe(recipe) {
  return {
    type: actionTypes.UPDATE_RECIPE,
    recipe,
  };
}

export function loadRecipe(recipe) {
  return {
    type: actionTypes.LOAD_RECIPE,
    recipe,
  };
}

export function getRecipeById(recipeId) {
  console.log(`${url}/${recipeId}`);
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/${recipeId}`);
      dispatch({
        type: actionTypes.LOAD_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPE_ERROR',
      });
    }
  };
}
