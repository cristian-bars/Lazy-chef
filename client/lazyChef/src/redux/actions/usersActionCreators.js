import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/users';
const url = 'http://192.168.0.29:2022';
//const url = process.env.REACT_APP_PORT;

export function loadUsers() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_USERS,
        users: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}

export function addUsers(user) {
  return {
    type: actionTypes.ADD_USER,
    user,
  };
}

export function deleteUser(userId) {
  return async dispatch => {
    await axios.delete(`${url}/${userId.id}`);
    dispatch({
      type: actionTypes.DELETE_USER,
      userId,
    });
  };
}

export function updateUser(user) {
  return {
    type: actionTypes.UPDATE_USER,
    user,
  };
}

export function loadUser(user) {
  return {
    type: actionTypes.LOAD_USER,
    user,
  };
}

export function getUserById(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/login`, user);
      dispatch({
        type: actionTypes.LOAD_USER,
        userToken: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}
