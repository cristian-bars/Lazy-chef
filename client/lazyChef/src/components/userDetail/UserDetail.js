/* eslint-disable react-hooks/exhaustive-deps */
import React, {useEffect} from 'react';
import {View, Text} from 'react-native';
import {connect} from 'react-redux';
import {PropTypes} from 'prop-types';
import {getUserById} from '../../redux/actions/usersActionCreators';
import styles from './recipeDetailStyles';

const UserDetail = ({user, dispatch, route}) => {
  const {recipeId} = route.params;
  useEffect(() => {
    dispatch(getUserById(recipeId));
  }, [recipeId]);

  return (
    <View style={styles.container}>
      <View>
        <Text style={styles.titleText}>{user.title}</Text>
      </View>
    </View>
  );
};

UserDetail.propTypes = {
  user: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(store) {
  return {
    user: store.user,
  };
}

export default connect(mapStateToProps)(UserDetail);
