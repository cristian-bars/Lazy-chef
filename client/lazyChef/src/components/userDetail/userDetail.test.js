import 'react-native';
import React from 'react';
import UserDetail from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import * as actions from '../../redux/actions/usersActionCreators';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a FavoritesList component', () => {
  const mockStore = Store([thunk]);
  let navigation;
  let myStore;
  let route;
  beforeEach(() => {
    myStore = mockStore({
      userAcces: {
        user: {
          email: 'Cristian@gmail.com',
          password: '1a',
          favouriteRecipes: ['4a3b2c1d'],
          recipesCreated: ['4a3b2c1d'],
        },
      },
      recipes: [
        {
          title: 'meat',
          _id: '4a3b2c1d',
          image: ['image1'],
          recipeIngredient: ['perejil', 'tomate'],
        },
        {
          _id: '123456',
          title: 'fish',
          image: ['image2'],
          recipeIngredient: ['agua', 'arroz'],
        },
      ],
    });
    navigation = {
      navigate: jest.fn(),
      goBack: jest.fn(),
    };
  });

  test('renders correctly', () => {
    const wrapper = render(
      <Provider store={myStore}>
        <UserDetail navigation={navigation} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
