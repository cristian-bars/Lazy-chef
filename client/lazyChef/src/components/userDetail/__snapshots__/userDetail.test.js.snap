// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a FavoritesList component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D3E0FF",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 10,
        "left": 20,
        "opacity": 1,
        "padding": 20,
        "position": "absolute",
        "top": 20,
        "width": 10,
        "zIndex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/back.png",
        }
      }
      style={
        Object {
          "height": 20,
          "left": -12,
          "position": "relative",
          "top": -10,
          "width": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D3E0FF",
        "display": "flex",
        "paddingTop": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#616161",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      ¡Hola 
      !
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/img/user.png",
        }
      }
      style={
        Object {
          "borderColor": "#FF914D",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 100,
          "margin": 10,
          "width": 100,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#616161",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Mis recetas
    </Text>
    <RCTScrollView
      data={
        Array [
          Object {
            "_id": "4a3b2c1d",
            "image": Array [
              "image1",
            ],
            "recipeIngredient": Array [
              "perejil",
              "tomate",
            ],
            "title": "meat",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 5,
                  "justifyContent": "space-between",
                  "margin": 5,
                  "paddingLeft": 10,
                  "paddingVertical": 10,
                  "width": 350,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#616161",
                    "fontSize": 20,
                    "maxWidth": 250,
                    "textAlign": "center",
                  }
                }
              >
                meat
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/img/trash.png",
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "marginRight": 20,
                    "width": 20,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 1,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Cochin",
            "fontSize": 25,
          }
        }
      >
        Añadir receta
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 1,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Cochin",
            "fontSize": 25,
          }
        }
      >
        Desconectar
      </Text>
    </View>
  </View>
</View>
`;
